From c0275ab189d619c2f688a383e5e83183f02eaf6f Mon Sep 17 00:00:00 2001
From: Donald Sharp <sharpd@nvidia.com>
Date: Tue, 10 Jan 2023 08:36:50 -0500
Subject: [PATCH] zebra: Continue fpm_read when we decide a netlink message is
 not needed

When FRR receives a netlink message that it decides to stop parsing
it returns a 0 ( instead of a -1 ).  Just make the dplane continue
reading other data instead of aborting the read.

Signed-off-by: Donald Sharp <sharpd@nvidia.com>
---
 zebra/dplane_fpm_nl.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/zebra/dplane_fpm_nl.c b/zebra/dplane_fpm_nl.c
index a31e79212..9d9ea7283 100644
--- a/zebra/dplane_fpm_nl.c
+++ b/zebra/dplane_fpm_nl.c
@@ -605,7 +610,10 @@ static void fpm_read(struct thread *t)
 				    hdr, 0, false, ctx) != 1) {
 				dplane_ctx_fini(&ctx);
 				stream_pulldown(fnc->ibuf);
-				return 0;
+				/*
+				 * Let's continue to read other messages
+				 * Even if we ignore this one.
+				 */
 			}
 			break;
 		default:
